rogue/src/components/status_effects.rs

```diff
@@ -112,25 +133,39 @@ pub fn tick_status_with_immunity<Status, StatusImmune>(
     }
 }
 
+pub fn remove_status<Status: Component + StatusEffect>(
+    store: &mut WriteStorage<Status>,
+    entity: Entity,
+) {
+    let status = store.get_mut(entity);
+    if let Some(_) = status {
+        store.remove(entity);
+    }
+}
+
+
 //------------------------------------------------------------------
 // Status Effect Components.
 //------------------------------------------------------------------
 pub trait StatusEffect {
-    fn new(turns: i32) -> Self;
+    fn new(turns: i32, render_glyph: bool) -> Self;
     fn remaining_turns(&self) -> i32;
     fn set_remaining_turns(&mut self, turns: i32);
+    fn do_render(&self) -> bool;
```

Comment:
This makes me wonder if it would be worth also associating the actual `StatusIndicatorGlyph` definition here. The `get_status_indicators(..)` function in `status_system.rs` currently holds that logic. Moving it into the trait would make it easier to generalize that function to just loop through a list of statuses and render the associated glyph.

Maybe a good exercise if more statuses get added later.

Reply:
Oh, yah, this is pretty obviously the right thing to do.